### PR TITLE
Nar 1528 restructure

### PR DESCRIPTION
**Pull request by erik-gbg**
NAR-1528
1. FLASK_APP values are changed. Set to "jobsearch" or "jobstream".
2.  **(no longer true)** _But to test the ontology loading you need to run docker, since for a localhost the ontlology loading is always delayed (which it has been since long ago)._

### CODE REVIEW

File: sokannonser/repository/__init__.py [hunk 1]
```diff
@@ -13,13 +13,22 @@
 
 
 class ElasticClient(object):
```
Comment (oskardrenske-ith) [inline sokannonser/repository/__init__.py:15]: can the class ElasticClient be moved to a separate file instead of having it in __init__.py ?

Comment (erik-gbg) [reply sokannonser/repository/__init__.py]: Done

**Review by oskardrenske-ith — APPROVED** [review]

**Comment by erik-gbg** [comment]
I want to get rid of this hack too. Anyone against?
https://github.com/JobtechSwe/sokannonser-api/blob/adddd2820a6ff509dc9b2050e8bd420e7dffc4ad/sokannonser/repository/text_to_concept.py#L28
if settings.ES_HOST != '127.0.0.1':

**Comment by oskardrenske-ith** [comment]
> I want to get rid of this hack too. Anyone against?
> 
> https://github.com/JobtechSwe/sokannonser-api/blob/adddd2820a6ff509dc9b2050e8bd420e7dffc4ad/sokannonser/repository/text_to_concept.py#L28

go for it!

**Review by lina1101 — APPROVED** [review]

**Comment by erik-gbg** [comment]
I will wait with the merger until after the Show & Tell

**Review by mrunesson — COMMENTED** [review]
Dockerfile must use a newer base image. README says it requires Py 3.9.4. Might be a good idea to take over the Dockerfile from castaway-experiment to reduce complexity.

In README, it would be valuable to describe difference of JobSearch and Jobstream. (external viewers have no idea)

Lot of nice cleaning!!!

All comments might not be relevant to do in this PR, but wanted to mention ones in my head.

**Comment by erik-gbg** [comment]
Good feedback, Magnus!

You decide what to do with the Dockerfile now. It's your call.

**Comment by erik-gbg** [comment]
Johan,

Can you describe the difference between JobSearch and Jobstream for README.md?

**Comment by mrunesson** [comment]
> Good feedback, Magnus!
> 
> You decide what to do with the Dockerfile now. It's your call.

I think this shall work and makes it easier to understand.

```
FROM  docker.io/library/python:3.9.7-slim-buster
EXPOSE 8081

RUN apt-get update -y && \
    apt-get install -y git build-essential && \
    apt-get clean -y

COPY . /app

WORKDIR /app

ENV TZ=Europe/Stockholm
RUN python3 -m pip install --upgrade setuptools wheel pip && \
    pip3 install -r requirements.txt && \
    python3 -m pytest -svv tests/unit_tests && \
    find tests -type d -name __pycache__ -prune -exec rm -rf -vf {} \;

USER 10000
CMD  uwsgi --http :8081 --manage-script-name --mount /=sokannonser:app
``` 

A new dependency is needed: uWSGI

nginx.conf, supervisord.conf, uwsgi.ini can all be removed.

**Though** memcached is removed in this setup, is that OK?

**Comment by erik-gbg** [comment]
Magnus,

That all looks great. But we should make a different branch/PR for this, and also coordinate it with elastic-importers.

And I cannot speak for memcached.

Also, --mount /=sokannonser:app looks old ;-)

**Comment by bryjo** [comment]
@erik-gbg @mrunesson Yes I'll update with the. difference of the 2 apps in the readme

File: sokannonser/repository/elastic_connection_with_retries.py [hunk 2]
```diff
@@ -48,6 +48,7 @@ def elastic_scan_with_retry(client, query, index, size, source):
 
 
 def create_elastic_client_with_retry(host, port=settings.ES_PORT, user=settings.ES_USER, password=settings.ES_PWD):
```
Comment (mattiasinsweden) [inline sokannonser/repository/elastic_connection_with_retries.py:50]: Couldn't this method move into the class/file sokannonser/repository/main_elastic_client.py ? (so file elastic_connection_with_retries.py can be deleted)

Comment (erik-gbg) [reply sokannonser/repository/elastic_connection_with_retries.py:50]: No, since it's used also by the ontology

**Pull request merged**